### PR TITLE
SDK update to 1.3.1, selectors update, added proxy usage, maxItems to…

### PR DESCRIPTION
**Pull request by OlehVeselov92**
… the input, better logging, new fields in the output

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                    for (const item of data) {
+                        if (!(itemsCounter >= maxItems)) {
+                            await Apify.pushData(item);
+                            itemsCounter += 1;
+                        }
+                    }
+                }
+            } else {
+                // University detail page
+                data = await page.$$eval('body', detailPageFunction);
```
Comment (pocesar) [inline src/main.js]: this is returning an array, the output is being `{ "0": {} }`. you can split the details when doing pushData (below)

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                data = {
+                    ...data,
+                    detail_page: url,
+                };
+                await Apify.pushData(data);
```
Comment (pocesar) [inline src/main.js]:
```js
// either one per row
for (const data of await page.$$eval('body', detailPageFunction)) {
   await Apify.pushData({ ...data, detail_page: url });
}
// or passed as an array to the output, `data` will be an array
data = await page.$$eval('body', detailPageFunction);
await Apify.pushData({ data, detail_page: url });
```

File: src/scrapers.js [hunk 3]
```diff
@@ -6,10 +6,13 @@ exports.pageFunction = ($universities) => {
     // We're getting the title, rank and country and link to detail of each university.
     $universities.forEach(($university) => {
         data.push({
-            title: $university.querySelector('.uni .title').innerText,
-            rank: $university.querySelector('.rank .rank').innerText,
-            country: $university.querySelector('.country > div').innerText,
-            detailPage: $university.querySelector('.uni .title').href,
+            title: $university.querySelector('.uni-link') ? $university.querySelector('.uni-link').textContent.trim() : null,
+            QS_World_University_Rankingnk: $university.querySelector('._univ-rank')
```
Comment (pocesar) [inline src/scrapers.js]: `Rankingnk` typo

File: src/scrapers.js [hunk 4]
```diff
@@ -22,23 +25,34 @@ exports.detailPageFunction = ($universities) => {
...
-        // University stats
-        uniStatsSelector.forEach((item) => {
-            const label = item.querySelector('label').innerText.replace(/\s/g, '');
-            const formattedLabel = label.charAt(0).toLowerCase() + label.slice(1);
-            const value = item.querySelector('.val').innerText;
-            result[formattedLabel] = value;
+        const locations = $university.querySelectorAll('.desktop-view .campus-locations');
+        const locationsResult = [];
+        locations.forEach((locationNode) => {
+            locationsResult.push(locationNode.textContent.trim());
```
Comment (pocesar) [inline src/scrapers.js]: this is having a "\n" inside, maybe split between address/country? there are two spans inside. the array can be like 

```js
locations.forEach((locationNode) => {
  const [location, country] = locationNode.querySelectorAll('span');
  locationsResult.push({
    country: country?.textContent?.trim(),
    mapLink: locationNode.href,
    location: location?.textContent?.trim().replace(/,$/, ''),
  });
});
```

**Review by pocesar — CHANGES_REQUESTED** [review]
minor changes

**Review by vbartonicek — APPROVED** [review]
Thanks for fixing that, @OlehVeselov92! I owe you a beer! :-)

**Review by pocesar — APPROVED** [review]

**Pull request merged**